### PR TITLE
Keycloak client role scope

### PR DESCRIPTION
**Pull request by desand01**
##### SUMMARY
New Keycloak module for client roles scope administration

##### ISSUE TYPE
New Module/Plugin Pull Request

##### COMPONENT NAME
keycloak_client_rolescope

##### ADDITIONAL INFORMATION
Role scope is used to control access to other clients' roles instead of using the 'Full scope allowed'. It's more secure to only send resource roles relevant to the current application and as the number of clients with their roles increases, some users might begin to have a problem with access tokens that are too large to be used as Bearer tokens.

### CODE REVIEW

File: plugins/modules/keycloak_client_rolescope.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    client_id:
+        type: str
+        required: true
+        description:
+            - Roles provided in O(role_names) while be added to this client scope.
+
+    client_scope_id:
+        type: str
+        description:
+            - If the O(role_names) are client role, the client id under which it resides.
```
Comment (felixfontein) [inline plugins/modules/keycloak_client_rolescope.py:65]:
```suggestion
            - If the O(role_names) are client role, the client ID under which it resides.
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! This looks pretty polished, I just noticed one small thing in the docs :)

File: plugins/modules/keycloak_client_rolescope.py (hunk 1 above)
Comment (desand01) [reply plugins/modules/keycloak_client_rolescope.py]: wow that was fast, thanks

File: plugins/modules/keycloak_client_rolescope.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+version_added: 8.6.0
+
+description:
+    - This module allows you to add or remove Keycloak roles from clients scope via the Keycloak REST API.
+      It requires access to the REST API via OpenID Connect; the user connecting and the client being
+      used must have the requisite access rights. In a default Keycloak installation, admin-cli
+      and an admin user would work, as would a separate client definition with the scope tailored
+      to your needs and a user having the expected roles.
+
+    - Client O(client_id) must have full_scope_allowed set to V(false).
```
Comment (felixfontein) [inline plugins/modules/keycloak_client_rolescope.py:26]: You can use
```suggestion
    - Client O(client_id) must have O(ignore:full_scope_allowed) set to V(false).
```
if that is not a real option, or
```suggestion
    - Client O(client_id) must have O(community.general.keycloak_client#module:full_scope_allowed) set to V(false).
```
to reference the `full_scope_allowed` option of the `community.general.keycloak_client` module.

File: plugins/modules/keycloak_client_rolescope.py (hunk 1 above)
Comment (felixfontein) [reply plugins/modules/keycloak_client_rolescope.py]: I just happened to look a minute after you created this PR ;)

**Comment by felixfontein** [comment]
Looks good to me. If nobody objects, I'll merge this ~tomorrow for the next release.

**Comment by felixfontein** [comment]
@desand01 thanks for your contribution!

**Pull request merged**